### DISCUSSION TITLE
Expected behavior of homogeneous to heterogeneous conversion of empty graphs

### DISCUSSION THREAD

**Discussion by ekosman** [pyg-team/pytorch_geometric#6295] [Q&A]
Hi,

I was facing an issue with the conversion of homogeneous data to heterogeneous data that represent an empty graph.

When trying to convert an empty homogeneous graph into a heterogeneous graph, is it expected that the `NodeStorage` contains an `edge_index` attribute? As far as I checked, if the homogeneous graph is initialized with an empty tensor, i.e. `edge_index=torch.tensor([[],[]])`, the object that I get after converting it into a `HeteroData` object using the `to_heterogeneous()` method contains `edge_index` in the `NodeStorage` referenced by the sole node type.

Example:
```
import torch
from torch_geometric.data import Data

d = Data(x=torch.randn(5,10), edge_index=torch.tensor([[],[]]))
hd = d.to_heterogeneous()
sole_node_type = hd.node_types[0]
has_edge_index = "edge_index" in hd[sole_node_type]
print(f"Does node {sole_node_type} has the edge_index attribute: {has_edge_index}")
print(f"Still, there are {len(hd.edge_types)} edge types in the heterogeneous graph")
```

For reference, if we add an edge to the graph the behavior changes:
```
import torch
from torch_geometric.data import Data

d = Data(x=torch.randn(5,10), edge_index=torch.tensor([[1],[1]]))
hd = d.to_heterogeneous()
sole_node_type = hd.node_types[0]
has_edge_index = "edge_index" in hd[sole_node_type]
print(f"Does node {sole_node_type} has the edge_index attribute: {has_edge_index}. (should be False)")
print(f"Now, there is {len(hd.edge_types)} edge types in the heterogeneous graph (should be 1)")
```

In the first example, `hd[sole_node_type]` has an attribute called `edge_index`. In the second example, it doesn't. I expected the second behavior.

**Comment by rusty1s** [comment]
Yeah, this is indeed a bug, sorry about that. I fixed it in https://github.com/pyg-team/pytorch_geometric/pull/6304. If you don't want to fix it by installing from source, you can pass the `edge_type_names` argument to prevent this from happening.
